docs/userGuide/components/diagrams.md

```diff
@@ -0,0 +1,41 @@
+{% set title = "Diagram Components" %}
+{% set filename = "diagrams" %}
+<span id="title" class="d-none">{{ title }}</span>
+
+<frontmatter>
+  title: "User Guide: {{ title }}"
+  layout: userGuide.md
+  pageNav: 3
+</frontmatter>
+
+<span id="link" class="d-none">
+<md>[_User Guide → {{ title }}_]({{ filename }}.html)</md>
+</span>
+
+<include src="advanced.md#slots-info" />
+
+# Diagram Components
+
+<span id="overview" class="lead">
+
+The image components here provide **convenient syntax & styling abstractions** on top of raw HTML and Markdown images.
```

Comment:
Thanks for looking through this!

> Should the previous page of Using HTML, JavaScript, CSS be Using Components -> Advanced instead?

Fixed.

>Should we change this to diagram so that it is consistent with the title?

Indeed, I actually meant to change the title to `Image & Diagram Components` instead.

I'm not sure if there's a better way of organising images and diagrams as well or if this is fine; On one hand they're slightly unrelated, on the other hand adding one more subpage makes the site nav seem a little too long.

Reply:
> I'm not sure if there's a better way of organising images and diagrams as well or if this is fine; On one hand they're slightly unrelated, on the other hand adding one more subpage makes the site nav seem a little too long.

I think this is fine for now since the section on diagrams is quite small. Once we introduce more ways of generating diagrams then we can consider splitting them up?